enhancements/sig-architecture/12-addon-manager/README.md

```diff
@@ -0,0 +1,300 @@
+   the managed cluster addon namespace. The addon will mount the secret to get the hub kubeconfig
+   to connect with the hub cluster
+
+When the certificate of managed cluster addon is about to expire, the registration agent will send
+a request to rotate the certificate on the hub cluster, the registration hub controller will auto
+approve the certificate rotation request.
+
+Taking observability addon as an example, its `managedclusteraddons` will be like
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: observability
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-addon-observability
+  registrations:
+  - signerName: "kubernetes.io/kube-apiserver-client"
+  - signerName: "observability-signer"
+    subject:
+      user: open-cluster-management-observability-cluster1
+      groups:
+      - open-cluster-management-observability
+      extras:
+        organizationalUnit: acm
+```
+
+In this example, the observability addon requires two csrs. One to access hub kube-api (with singer name
+`kubernetes.io/kube-apiserver-client`), and another to access its own endpoint (with singer name
+`observability-signer`). It depends on its own hub addon controller to decide whether the csr
+should be approved or not.
+
+### Test Plan
+
+#### Unit Tests
+
+All controller implementations will have thorough coverage of unit tests.
+
+#### Integration Tests
+Considering certificate operation, we will use integration tests to coverage addon registration
+test cases, includes:
+
+- Register an addon with kube singer
+- Register an addon with custom signer
+- Update an addon registration configuration
+- Addon client certification rotation
+
+#### E2E Tests
+
+Using e2e tests to coverage addon lease update test cases, includes:
+
+- Maintain addon status to available if addon update its lease constantly
+- Update addon status to unavailable if addon stops to update its lease
+- Update addon status to unknown if there is no lease for this addon
+- Update addon status to unknown if managed cluster becomes to unknown
+
+### Graduation Criteria
+
+#### GA Graduation
+
+- All design are implemented
+- Address known bugs and add tests to prevent regressions
+- Docs are up-to-date with latest implemented
+
+### Upgrade Strategy
+
+- The `ManagedClusterAddOn` API should be updated on the hub cluster
+- For lease update, each addon needs to create and update its lease on its installation namespace
+  on the managed cluster
+- For each addon, if it needs to register to hub cluster, it needs to leverage addon framework to
+  recode its registration configuration information to the its `ManagedClusterAddOn` CR
+
+### Version Skew Strategy
+
+This design will be implemented on ACM 2.3
+
+For addon registration part, it is a new feature, it will not affect the addons of the old version.
+
+For lease update part, it will keep compatibility with 2.2, and after 2.3, the compatibility of 2.2
```

Comment:
one thing we have not considered is the new hub with old  klusterlet/old agent. In this case, the addon status will be unavailable?

Reply:
there will be no available status for addons